include/wil/win32_helpers.h

```diff
@@ -887,6 +887,177 @@ bool init_once(_Inout_ INIT_ONCE& initOnce, T func)
         return false;
     }
 }
+
+namespace details
+{
+    // An iterator that can be used to enumerate processes in the system.
+    // Usage:
+    // for (auto process : process_iterator(L"ProcessName.exe"))
+    // {
+    //     // Do something with process, e.g.
+    //     // wil::unique_handle hProcess(::OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, process.th32ProcessID));
+    // }
+    struct process_entry
+    {
+        DWORD ProcessID{};
+        ULONG_PTR ThreadCount{};
+        DWORD ParentProcessID{};
+        LONG BasePriority{};
+        std::wstring ProcessName;
+
+        process_entry(std::wstring_view processName = L"") :
+            m_processName(!processName.empty() ? ::wil::make_unique_string<::wil::unique_cotaskmem_string>(processName.data()) : nullptr),

```

Comment:
String views are not null terminated, so `.data()` here is UBER SUPER MEGA DANGER.  Maybe use `wil::zwstring_view` which requires its source to be a null-terminated thing?

Reply:
changed to zwstring (moved to a new file to not pollute win32_helpers too much with includes)